homeassistant/components/fitbit/config_flow.py

```diff
@@ -53,6 +53,21 @@ async def async_step_reauth_confirm(
             return self.async_show_form(step_id="reauth_confirm")
         return await self.async_step_user()
 
+    async def async_step_creation(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Create config entry from external data with Fitbit specific error handling."""
+        try:
+            return await super().async_step_creation()
+        except FitbitAuthException as err:
+            _LOGGER.error(
+                "Failed to authenticate when creading Fitbit credentials: %s", err
+            )
+            return self.async_abort(reason="invalid_access_token")
```

Comment:
I don't think this is completely correct. We're trying to fetch the tokens here, so we haven't received those yet when this error happens. It's more a broader case of invalid authentication, I think. When testing this I had reset the client secret in the application without updating the secret in application credentials.

Reply:
```suggestion
            return self.async_abort(reason="invalid_auth")
```